envoy/server/options.h

```diff
@@ -266,6 +266,11 @@ class Options {
    * responsibility of the caller to handle the duplicates.
    */
   virtual const Stats::TagVector& statsTags() const PURE;
+
+  /**
+   * @return the type of listener manager to create.
+   */
+  virtual const std::string& listenerManager() const PURE;
```

Comment:
Can this return an enum instead of a string where the valid value space is ambiguous?

Reply:
We can, but I don't think Envoy should have to know the name of the E-M listener, and this allows other to do overrides, so I think the string is appropriate.

Reply:
ok